sdk/nodejs/clusterMixins.ts

```diff
@@ -67,7 +67,7 @@ Object.defineProperty(Cluster.prototype, 'provider', {
         if (!this._provider) {
             this._provider = new Provider(`${this.__name}-provider`, {
                 kubeconfig: this.kubeconfigJson,
-            }, { parent: this, aliases: [{ name: this.__name }]});
+            }, { parent: this });
```

Comment:
This needs the alias otherwise a new provider will be created and the old one destroyed. The alias lets pulumi know that even though the name has changed, they are the same.